File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
  const sum = nums.reduce((acc, cur) => acc + cur, 0);
  let leftSum = 0;
  let rightSum = sum;
  
  for (let i = 0; i < nums.length; i++) {
    const cur = nums[i];

    rightSum -= cur;

    if (leftSum === rightSum) {
      return i;
    }
    
    leftSum += cur;
  }
  
  return -1;
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
  def pivotIndex(self, nums: List[int]) -> int:
    total_sum = reduce(lambda x, y: x + y, nums, 0)
    cur_sum = 0

    for i in range(0, len(nums)):
      cur = nums[i]
      cur_sum += cur

      if (total_sum - cur) / 2 == (cur_sum - cur):
        return i

    return -1
